### PR TITLE
keys: change to []bytes for keys

### PR DESCRIPTION
**Pull request by marbar3778**
- changes the type of keys from [#]bytes to []bytes
- change bytes methods to return byte form of the jey not amino encoding

ref #3739 

______

For contributor use:

- [ ] Wrote tests
- [ ] Updated CHANGELOG_PENDING.md
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Updated relevant documentation (`docs/`) and code comments
- [ ] Re-reviewed `Files changed` in the Github PR explorer

### CODE REVIEW

File: crypto/encoding/amino/encode_test.go [hunk 1]
```diff
@@ -17,18 +16,21 @@ import (
...
 
 func checkAminoBinary(t *testing.T, src, dst interface{}, size int) {
 	// Marshal to binary bytes.
 	bz, err := cdc.MarshalBinaryBare(src)
 	require.Nil(t, err, "%+v", err)
-	if byterSrc, ok := src.(byter); ok {
-		// Make sure this is compatible with current (Bytes()) encoding.
-		assert.Equal(t, byterSrc.Bytes(), bz, "Amino binary vs Bytes() mismatch")
-	}
+	// if byterSrc, ok := src.(AminoMarshal); ok {
```
Comment (marbar3778) [inline crypto/encoding/amino/encode_test.go:28]: will uncomment this in a separate PR.

**Comment by alexanderbez** [comment]
This is being split across multiple PRs piecemeal, correct @marbar3778?

**Comment by marbar3778** [comment]
bad rebase, now its fixed. This is the broken out pr, sorry about that

File: crypto/secp256k1/secp256k1_test.go [hunk 2]
```diff
@@ -36,15 +36,19 @@ func TestPubKeySecp256k1Address(t *testing.T) {
...
 
-		var priv secp256k1.PrivKey
-		copy(priv[:], privB)
+		var priv secp256k1.PrivKey = privB
 
 		pubKey := priv.PubKey()
 		pubT, _ := pubKey.(secp256k1.PubKey)
-		pub := pubT[:]
-		addr := pubKey.Address()
+		// var priv secp256k1.PrivKey
```
Comment (melekes) [inline crypto/secp256k1/secp256k1_test.go:43]: will you uncomment this in another PR?

File: crypto/ed25519/ed25519.go [hunk 3]
```diff
@@ -118,46 +110,39 @@ func genPrivKey(rand io.Reader) PrivKey {
...
+// PubKeyEd25519 implements crypto.PubKey for the Ed25519 signature scheme.
+type PubKey []byte
 
 // Address is the SHA256-20 of the raw pubkey bytes.
 func (pubKey PubKey) Address() crypto.Address {
-	return crypto.Address(tmhash.SumTruncated(pubKey[:]))
+	if len(pubKey) != PubKeySize {
+		panic("pubkey is incorrect size")
+	}
+	return crypto.Address(tmhash.SumTruncated(pubKey))
```
[the remark this answers is not in the document]
Comment (marbar3778) [reply crypto/ed25519/ed25519.go:128]: I don't think it's necessary, it is an low overhead. There are other ways we can get greater performance on ed25519 keys.

File: crypto/secp256k1/secp256k1.go [hunk 4]
```diff
@@ -136,30 +116,30 @@ const PubKeySize = 33
...
-type PubKey [PubKeySize]byte
+type PubKey []byte
 
 // Address returns a Bitcoin style addresses: RIPEMD160(SHA256(pubkey))
 func (pubKey PubKey) Address() crypto.Address {
+	if len(pubKey) != PubKeySize {
+		panic("length of pubkey is incorrect")
+	}
+
 	hasherSHA256 := sha256.New()
```
[the remark this answers is not in the document]
Comment (marbar3778) [reply crypto/secp256k1/secp256k1.go:125]: this is more up to the sdk team as we won't be using it in our proto work. Also it is unclear if they will be setting this key type in the sdk directly or using the one defined here.

**Pull request merged**